pkg/controllers/jobset_controller.go

```diff
@@ -377,15 +426,47 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 			return err
 		}
 	}
-
 	var lock sync.Mutex
 	var finalErrs []error
-
-	for _, rjob := range js.Spec.ReplicatedJobs {
-		jobs, err := constructJobsFromTemplate(js, &rjob, ownedJobs)
+	allJobsCreatedForStartupPolicy := false
+	statusOfFirstJob := findReplicatedStatus(replicatedJobStatus, js.Spec.ReplicatedJobs[0].Name)
+	if !isJobSetSuspended(js) && inOrderStartupPolicy(js.Spec.StartupPolicy) && !replicatedJobsAtLeastReady(js.Spec.ReplicatedJobs[0].Replicas, statusOfFirstJob) {
+		err := r.ensureCondition(ctx, js, corev1.EventTypeNormal, true, generateStartupPolicyCondition(false))
+		if err != nil {
+			return fmt.Errorf("unable to update condition for startup policy: %v", err)
+		}
+	}
+	for idx, replicatedJob := range js.Spec.ReplicatedJobs {
+		jobs, err := constructJobsFromTemplate(js, &replicatedJob, ownedJobs)
 		if err != nil {
 			return err
 		}
+		if !isJobSetSuspended(js) {
+			if inOrderStartupPolicy(js.Spec.StartupPolicy) {
```

Comment:
pls merge the two if statements into one

Reply:
updated.